api/src/providers/demarcheSimplifiee/demarcheSimplifieeProvider.test.ts

```diff
@@ -0,0 +1,319 @@
+import { getUaiFromString, mapAppelAProjetDemarcheSimplifieeDtoToAppelAProjet } from "./demarcheSimplifieeProvider";
+import { AppelAProjetDemarcheSimplifieeDto } from "./appelAProjetDemarcheSimplifieeDto";
+
+describe("demarcheSimplifieeProvider", () => {
+  it("should split the input string correctly", () => {
+    const input = "Lycée, Ville (UAI42)";
+    const expectedOutput = "UAI42";
+
+    const result = getUaiFromString(input);
+
+    expect(result).toEqual(expectedOutput);
+  });
+
+  it("should return the right data", () => {
+    const mockAppelAProjetDto = getMockAppelAProjetDto();
+    const result = mapAppelAProjetDemarcheSimplifieeDtoToAppelAProjet(mockAppelAProjetDto);
+
+    const expectedAppelAProjet = [
+      {
+        etablissement: { uai: "SOME_UAI", email: "mail@etablissement.fr" },
+        classe: {
+          referent: { nom: "NOM_REFERENT", prenom: "PRENOM_REFERENT", email: "email@referent.fr" },
+          nom: "SNU 2024 2025",
+          nombreElevesPrevus: "42",
+          coloration: "Résilience et prévention des risques",
+          trimestre: "Au 3ème trimestre (séjour réalisé en avril, mai ou juin 2025).",
+        },
+      },
+    ];
+    expect(result).toEqual(expectedAppelAProjet);
+  });
+});
+
+const getMockAppelAProjetDto = (): AppelAProjetDemarcheSimplifieeDto => ({
```

Comment:
est-ce que ça ne vaudrait pas le cout de le déplacer dans une json ?

Reply:
Pourquoi pas?
Je vais devoir l'intégrer aussi côté __tests__ pour les IT et de ce côté j'aurais besoin d'une fonction pour passer des paramètres. Il faudra échanger sur le partage de mock en __test__ et src/